### DISCUSSION TITLE
Design doc for interfaces and issue list?

### DISCUSSION THREAD

**Discussion by rahul101001000** [Sunbird-RC/community#342] [Ideas & Enhancements]
@suresh12 @parthlawate @ChakshuGautam @amarts @girirajdaga @sukhpreetsamagra

For each of these building blocks being selected, can we ensure well defined interfaces in Sunbird RC as @pramodkvarma also called out in one of the threads. While we may be picking one building block for the ULP implementation, other implementations should have the option to choose a different one. 

Loosely coupled interfaces may be needed for - 
- auth for issuer (e.g. I should be able to replace Keycloak with Fusion Auth or custom auth)
- credential generation
- template management
- notifications 
- telemetry
- verification
- wallet
- key management 
- DLT like CORD

Questions - 
1. Is there a doc that lays out all these interfaces?
2. Is there a issue list that summarizes changes in base RC that would be needed enable these interfaces?

This may not be happening immediately but would be good to have a live wiki and a issue list that can be tracked over time. 

Thank you!

**Comment by ChakshuGautam** [comment]
@rahul101001000 Started with a thread on Consent here - https://github.com/orgs/Sunbird-RC/discussions/348
Will create a ticket where everything is tracked on the list.

**Comment by surendrasinghs** [comment]
Great suggestion @rahul101001000. Good to see various discussions coming together in one thread.



> 1. Is there a doc that lays out all these interfaces?

@tejash-jl can we ensure documentation has API specs for all the components of RC?
Based on this, we can come up with changes required. 

@ChakshuGautam Please share the ticket and any additional detail. Thanks!
